### ISSUE TITLE
[V3 Modlog] Modlog case creation with null modlog channel

### ISSUE THREAD

**Issue by aikaterna** [Cog-Creators/Red-DiscordBot#2220]
# Command bugs

#### Command name

Any command that's configured to write to the modlog on execution.

#### What cog is this command from?

Mod/Modlog

#### What were you expecting to happen?

For the RuntimeError in get_modlog_channel to be handled.

#### What actually happened?

```
[09/10/2018 21:58] ERROR events on_error 191: Exception in modlog_case_create
Traceback (most recent call last):
  File "C:\Program Files\Python37\lib\site-packages\discord\client.py", line 225, in _run_event
    await coro(*args, **kwargs)
  File "C:\Program Files\Python37\lib\site-packages\redbot\cogs\mod\mod.py", line 1594, in on_modlog_case_create
    mod_channel = await modlog.get_modlog_channel(case.guild)
  File "C:\Program Files\Python37\lib\site-packages\redbot\core\modlog.py", line 694, in get_modlog_channel
    raise RuntimeError("Failed to get the mod log channel!")
RuntimeError: Failed to get the mod log channel!
```
Also see my comment on #2164.

#### How can we reproduce this issue?

Have a new settings.json for the modlog or un-set an existing modlog channel then use an action that would normally be posted to the modlog like a ban or hackban.

**Comment by Kowlin** [comment]
PR Pending: #2331

**Comment by Tobotimus** [comment]
Resolved by #2331.
